public/app/features/dashboard-scene/scene/layout-tabs/TabsLayoutManager.tsx

```diff
@@ -0,0 +1,108 @@
+import { SceneObjectBase, SceneObjectState, VizPanel } from '@grafana/scenes';
+import { t } from 'app/core/internationalization';
+
+import { DashboardLayoutManager } from '../types/DashboardLayoutManager';
+
+import { TabItem } from './TabItem';
+import { TabsLayoutManagerRenderer } from './TabsLayoutManagerRenderer';
+
+interface TabsLayoutManagerState extends SceneObjectState {
+  tabs: TabItem[];
+  currentTab: TabItem;
+}
+
+export class TabsLayoutManager extends SceneObjectBase<TabsLayoutManagerState> implements DashboardLayoutManager {
+  public static Component = TabsLayoutManagerRenderer;
+
+  public readonly isDashboardLayoutManager = true;
+
+  public static readonly descriptor = {
+    get name() {
+      return t('dashboard.tabs-layout.name', 'Tabs');
+    },
+    get description() {
+      return t('dashboard.tabs-layout.description', 'Tabs layout');
+    },
+    id: 'tabs-layout',
+    createFromLayout: TabsLayoutManager.createFromLayout,
+  };
+
+  public readonly descriptor = TabsLayoutManager.descriptor;
+
+  public addPanel(vizPanel: VizPanel) {
+    this.state.currentTab.getLayout().addPanel(vizPanel);
+  }
+
+  public removePanel(panel: VizPanel) {
+    this.state.currentTab.getLayout().removePanel(panel);
+  }
+
+  public duplicatePanel(panel: VizPanel) {
+    this.state.currentTab.getLayout().duplicatePanel(panel);
+  }
+
+  public getVizPanels(): VizPanel[] {
+    const panels: VizPanel[] = [];
+
+    for (const tab of this.state.tabs) {
+      const innerPanels = tab.getLayout().getVizPanels();
+      panels.push(...innerPanels);
+    }
+
+    return panels;
+  }
+
+  public getMaxPanelId(): number {
+    return Math.max(...this.state.tabs.map((tab) => tab.getLayout().getMaxPanelId()));
+  }
+
+  public addNewRow() {
+    this.state.currentTab.getLayout().addNewRow();
+  }
+
+  public addNewTab() {
+    const currentTab = new TabItem();
+    this.setState({ tabs: [...this.state.tabs, currentTab], currentTab });
+  }
+
+  public editModeChanged(isEditing: boolean) {
+    this.state.tabs.forEach((tab) => tab.getLayout().editModeChanged?.(isEditing));
+  }
+
+  public activateRepeaters() {
+    this.state.tabs.forEach((tab) => tab.getLayout().activateRepeaters?.());
+  }
+
+  public removeTab(tab: TabItem) {
+    if (this.state.tabs.length === 1) {
+      throw new Error('TabsLayoutManager: Cannot remove last tab');
```

Comment:
should we throw? maybe just ignore the delete for now and show a console warning?

Reply:
🤔 let's revise this in a subsequent PR.

Reply:
example

https://github.com/user-attachments/assets/aa92a31c-2554-4643-96bd-56a215a4feae

Reply:
Fixed in current iteration.